group-coordinator/src/test/java/org/apache/kafka/coordinator/group/runtime/InMemoryPartitionWriter.java

```diff
@@ -102,6 +241,31 @@ public long append(
         }
     }
 
+    @Override
+    public long appendEndTransactionMarker(
+        TopicPartition tp,
+        long producerId,
+        short producerEpoch,
+        int coordinatorEpoch,
+        TransactionResult result
+    ) throws KafkaException {
+        PartitionState state = partitionState(tp);
+        state.lock.lock();
+        try {
+            state.entries.add(new LogControl(
+                producerId,
+                producerEpoch,
+                coordinatorEpoch,
+                result
+            ));
+            state.endOffset += 1;
+            if (autoCommit) commit(tp, state.endOffset);
```

Comment:
what does autoCommit mean here?

Reply:
It means that we advance the HWM immediately and trigger the `onHighWatermarkUpdated` callback if enabled.

Reply:
Is this like acks = 0? I wasn't aware of when this was used.

Reply:
No. It is more like having a partition with a single replica so the HWM advances directly. Keep in mind that this is designed for being used in tests.

Reply:
synced offline -- this is like acks=all for one replica and helpful in tests.